### ISSUE TITLE
[Compatibility] Conflict with DAE causes Max Spell Points to increase on cast

### ISSUE THREAD

**Issue by megahead11** [misthero/dnd5e-spellpoints#7]
Not sure why this is happening but I think it might be a change to something with DAE recently or with 1.2.2 DND 5e (i've not experienced it before)

However, if an effect from DAE increases the max spell points (for example, resources.primary.max + 10), casting a spell on an actor with spell points enabled and consuming the points will increase the maximum value by some strange amount. 

(Edit: might not just be DAE but active effects in general)

**Comment by megahead11** [comment]
Addendum, I realize the amount being increased is based on what's being added by dae. For example, if you have resource.primary.max + 10, the maximum value of the spell points increases by 10 each time. I assume the maximum value is being re-calculated and updated on each cast, causing this to happen.

**Comment by MrVauxs** [comment]
This is how generally Active Effects work, you'd have to use another syntax for it to be bug-less as it *will* change everytime you update it. Like how you change your AC while wearing armor it will update back to the set amount by the armor.

**Comment by megahead11** [comment]
Seeing as I didnt have dmg progression on, I don't think theres any reason it should be updating the max anyways.

**Comment by misthero** [comment]
the "calculateSpellPoints" funcion is the one responsible of changing the max spell point resource value, this funciont is called only on class item update and it does nothing if the DMG calculations is "off".

There must be something else going on here.

**Comment by megahead11** [comment]
I figured out the issue. On line 183/184, you have it updating the entire resource as opposed to only the value when processing the points after casting.
```
   update.data.resources = {
      [spellPointResource.key] : spellPointResource.values
    };
```
This particular block seems to update both the current and max spell points when it runs.
Instead, it should only update the value of the resource. 

I tried it using this block of code and it works as expected (but not efficiently because I literally copy-pasted from later down the code lol)

```
let updateActor = {[`data.resources.${spellPointResource.key}.value`] : spellPointResource.values.value};	
        actor.update(updateActor); 
```

I imagine you probably figured this out by now since it's been a while since this issue was posted
